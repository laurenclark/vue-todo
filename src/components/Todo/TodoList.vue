<template>
    <ul v-for="(todo, index) of todos" :key="index">
        <TodoListItem :todo="todo" />
    </ul>
</template>

<script>
import TodoListItem from "./TodoListItem.vue";
export default {
    name: "TodoForm",
    components: {
        TodoListItem
    },
    data() {
        return {
            todos: [...new Array(4)]
        };
    }
};
</script>

<style>
</style>