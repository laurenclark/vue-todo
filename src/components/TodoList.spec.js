import TodoList from "./TodoList.vue";
import { mount } from "@vue/test-utils";

describe("The right component is rendering", () => {
    it("Exports a valid component", () => {
        expect(TodoList).toBeAComponent();
    });
    it("Renders correctly", () => {
        const wrapper = mount(TodoList);
        expect(wrapper.element).toMatchSnapshot();
    });
});
