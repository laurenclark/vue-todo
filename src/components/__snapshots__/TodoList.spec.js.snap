// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The right component is rendering Renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <ul>
    <li>
      <span>
        List Item 
        <span>
          Thu Jan 06 2022 00:38:50 GMT+0000 (Greenwich Mean Time)
        </span>
        <span>
          Delete
        </span>
      </span>
    </li>
  </ul>
  <ul>
    <li>
      <span>
        List Item 
        <span>
          Thu Jan 06 2022 00:38:50 GMT+0000 (Greenwich Mean Time)
        </span>
        <span>
          Delete
        </span>
      </span>
    </li>
  </ul>
  <ul>
    <li>
      <span>
        List Item 
        <span>
          Thu Jan 06 2022 00:38:50 GMT+0000 (Greenwich Mean Time)
        </span>
        <span>
          Delete
        </span>
      </span>
    </li>
  </ul>
  <ul>
    <li>
      <span>
        List Item 
        <span>
          Thu Jan 06 2022 00:38:50 GMT+0000 (Greenwich Mean Time)
        </span>
        <span>
          Delete
        </span>
      </span>
    </li>
  </ul>
  
</div>
`;

exports[`is the TodoList Component exports a valid component 1`] = `
Object {
  "components": Object {
    "TodoListItem": Object {
      "name": "TodoListItem",
      "render": [Function],
    },
  },
  "data": [Function],
  "name": "TodoForm",
  "render": [Function],
}
`;
